# HG changeset patch
# User jwe
# Date 1201511599 0
#      Mon Jan 28 09:13:19 2008 +0000
# Branch release-3-0-x
# Node ID d00dcbc03a4cfecd267d42d0d7a76b11abd2480a
# Parent  fe5250e05ceb59a5c3763da63fbe9c75d2c7f6b2
[3-0-0-branch @ 2008-01-28 09:13:19 by jwe]

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-01-28  Michael Goffioul <michael.goffioul@gmail.com>
+
+	* plot/xlabel.m, plot/ylabel.m, plot/zlabel.m:
+	Return the label handle, not the current axis handle.
+
 2008-01-25  James Packer  <james.packer@wadh.ox.ac.uk>
 
 	* geometry/griddata3.m: Call griddata, not gridata.
 
 2008-01-24  Schloegl Alois  <alois.schloegl@tugraz.at>
 
 	* linear-algebra/trace.m: Require 2-d args.  Handle vectors properly.
 
diff --git a/scripts/plot/xlabel.m b/scripts/plot/xlabel.m
--- a/scripts/plot/xlabel.m
+++ b/scripts/plot/xlabel.m
@@ -44,12 +44,12 @@ function retval = xlabel (varargin)
   unwind_protect
     axes (h);
     tmp = __axis_label__ ("xlabel", varargin{:});
   unwind_protect_cleanup
     axes (oldh);
   end_unwind_protect
 
   if (nargout > 0)
-    retval = h;
+    retval = tmp;
   endif
 
 endfunction
diff --git a/scripts/plot/ylabel.m b/scripts/plot/ylabel.m
--- a/scripts/plot/ylabel.m
+++ b/scripts/plot/ylabel.m
@@ -39,12 +39,12 @@ function retval = ylabel (varargin)
   unwind_protect
     axes (h);
     tmp = __axis_label__ ("ylabel", varargin{:});
   unwind_protect_cleanup
     axes (oldh);
   end_unwind_protect
 
   if (nargout > 0)
-    retval = h;
+    retval = tmp;
   endif
 
 endfunction
diff --git a/scripts/plot/zlabel.m b/scripts/plot/zlabel.m
--- a/scripts/plot/zlabel.m
+++ b/scripts/plot/zlabel.m
@@ -39,12 +39,12 @@ function retval = zlabel (varargin)
   unwind_protect
     axes (h);
     tmp = __axis_label__ ("zlabel", varargin{:});
   unwind_protect_cleanup
     axes (oldh);
   end_unwind_protect
 
   if (nargout > 0)
-    retval = h;
+    retval = tmp;
   endif
 
 endfunction
